File: src/compiler/typescript_to_rust/generators/ic_object/functions/stable_size.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionStableSize(): Rust {
    return /* rust */ `
        fn _azle_ic_stable_size(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::stable::stable_size().azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate_rearchitecture/src/ic/stable_size.rs
use proc_macro2::TokenStream;
use quote::quote;

pub fn generate() -> TokenStream {
    quote! {
        fn stable_size<'a>(
            context: &'a JSContextRef,
            _this: &CallbackArg,
            _args: &[CallbackArg],
        ) -> Result<JSValueRef<'a>, anyhow::Error> {
            let return_js_value: JSValue =
                candid::encode_one(ic_cdk::api::stable::stable_size())?.into();
            to_qjs_value(&context, &return_js_value)
        }
    }
}
